### PR TITLE
Ensure `x-rh-identity` header `Role` attribute is a list

### PR DESCRIPTION
**Pull request by coderbydesign**
Since our SAML assertion sends `Role` as a multi-value attribute like:
```
{
  "Role": ["Role1", "Role2"],
  "givenName": ["Jane"]
}
```
Our identity header would provide you with:
```
{
  "Role": ["Role1", "Role2"],
  "givenName": "Jane"
}
```
This will preserve the role list, and unpack other values with one element.

There is a case where `Role` may contain just one role, like:
```
{
  "Role": ["Role1"]
}
```

In this case, the current logic would unpack that as well, and we'd end up with
a header like:
```
{
  "Role": "Role1",
  "givenName": "Jane"
}
```

This chance will maintain a list for `Role` regardless of the number of elements.
Since this was a customization on the SAML assertion from the IAM team, we should
be able to rely on other fields always being single-value or multi-value, though
we may want to confirm this.

Other minor changes:
- comments for CI deployment
- pre-populate `.env.example` with `FLASK_ENV=development` for debugging

### CODE REVIEW

File: services/web/app.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def auth_saml(self, auth):
             if authorized:
                 resp = make_response("Authorized", 200)
                 resp.headers["X-RH-Identity"] = self.make_identity_header(
-                    "Associate", "saml-auth", {k: v if len(v) > 1 else v[0] for k, v in auth_data}
+                    "Associate", "saml-auth", {k: v if (len(v) > 1 or k == "Role") else v[0] for k, v in auth_data}
```
Comment (j00bar) [inline services/web/app.py:181]: I feel like we should have an enumeration of multi-value attributes, no?

Comment (coderbydesign) [reply services/web/app.py]: Happy to rebase on #12 when it goes in, but I updated this PR.

File: services/web/app.py [hunk 2]
```diff
@@ -175,10 +175,12 @@ def auth_saml(self, auth):
             app.logger.debug(f"SAML auth_data: {auth_data}")
             predicate = auth["saml"]
             authorized = eval(predicate, dict(user=auth_data))
+            multi_value_attrs = os.environ.get("MULTI_VALUE_SAML_ATTRS", "").split(",")
```
Comment (j00bar) [inline services/web/app.py:178]: Er, this probably belongs in `config.py`, not the environment :)

Comment (coderbydesign) [reply services/web/app.py:178]: Blast! That's where I had it initially, and then I got in my own head. 😆 I'll move it back, thanks!

**Review by j00bar — APPROVED** [review]
Go ahead and merge yours - I'll rebase off of the changes.

**Pull request merged**